profile/profile.js

```diff
@@ -0,0 +1,361 @@
+		}
+	});
+};
+
+let loadAbilities = function() {
+	$.ajax({
+		url: "/server/profile/profile.php",
+		type: "GET",
+		dataType: "JSON",
+		data: {
+			action: 'getAbilities'
+		},
+		cache: false,
+		success: function(response) {
+			$('#abilitiesSelect').find('option').remove();
+			for (let i = 0; i < response.abilities.length; i++) {
+				let ability = response.abilities[i];
+				let modifiers = ability.has ? `data-userAbilityId=${ability.userAbilityId} selected` : '';
+				let option = $(`<option value=${ability.abilityId} ${modifiers}>${ability.name}</option>`);
+				$('#abilitiesSelect').append(option);
+			}
+			$('#abilitiesSelect').selectpicker();
+
+			$('#certificationsDiv').find('div').remove();
+			for (let i = 0; i < response.abilitiesRequiringVerification.length; i++) {
+				let ability = response.abilitiesRequiringVerification[i];
+				
+				let abilityDiv = $('<div></div>');
+				let abilityName = $('<span></span>').html(ability.name);
+				let abilityStatus = $('<i></i>').addClass(`${ability.has ? 'fa fa-check icon green-icon' : 'fa fa-times icon red-icon'}`);
+
+				abilityDiv.append(abilityName, abilityStatus);
+				$('#certificationsDiv').append(abilityDiv);
+			}
+		},
+		error: function(response) {
+			alert("Unable to load abilities. Please refresh the page in a few minutes.");
+		}
+	});
+};
+
+// Maps a siteId -> dates that site is open -> shifts for that date
+let shiftsMap = new Map();
+// Maps a siteId to the title of the site
+let sitesMap = new Map(); 
+
+let loadShifts = function() {
+	$.ajax({
+		url: "/server/profile/profile.php",
+		type: "GET",
+		dataType: "JSON",
+		data: {
+			action: 'getShifts'
+		},
+		cache: false,
+		success: function(response) {
+			for (let i = 0; i < response.shifts.length; i++) {
+				let shift = response.shifts[i];
+
+				// Create the shiftsMap
+				if (!sitesMap.has(shift.siteId)) sitesMap.set(shift.siteId, shift.title);
+				if (!shiftsMap.has(shift.siteId)) shiftsMap.set(shift.siteId, new Map());
+				
+				let startDateTime = new Date(shift.startTime + ' CST');
+				let endDateTime = new Date(shift.endTime + ' CST');
+
+				let date = startDateTime.toLocaleDateString();
+
+				let datesMap = shiftsMap.get(shift.siteId);
+				if (!datesMap.has(date)) datesMap.set(date, []);
+
+				let startTimeString = getTimeString(startDateTime);
+				let endTimeString = getTimeString(endDateTime);
+
+				let shiftTimes = datesMap.get(date);
+				shiftTimes.push({
+					'shiftId': shift.shiftId,
+					'startTime': startTimeString,
+					'endTime': endTimeString,
+					'signedUp': shift.signedUp
+				});
+
+				// Append any shifts the person is already signed up for
+				if (shift.signedUp) {
+					appendSignedUpShift(shift.title, date, startTimeString, endTimeString, shift.userShiftId);
+				}
+			}
+		},
+		error: function(response) {
+			alert("Unable to load shift. Please refresh the page in a few minutes.");
+		}
+	});
+}
+
+let getTimeString = function(dateTime) {
```

Comment:
With this logic on the server-side, check out [this link](http://www.eltcalendar.com/stuff/datemysqlphp.html) for formatting. It's the most helpful list I've found.